spec/nodeattr_interface_spec.rb

```diff
@@ -27,6 +27,23 @@
 require 'exceptions'
 
 RSpec.describe Metalware::NodeattrInterface do
+  context 'with setup1 genders' do
+    before do
+      SpecUtils.use_mock_genders(self, genders_file: 'setup1/genders')
+    end
+
+    describe '#nodes_in_primary_group' do
+      it 'returns only the nodes in the primary group' do
+        nodes = Metalware::NodeattrInterface.nodes_in_primary_group('group1')
+        nodes_not_in_primary_group = nodes.select do |node|
+          node.match(/\AnodeA\d\d\Z/).nil?
+        end
+        expect(nodes.length).to eq(10)
+        expect(nodes_not_in_primary_group.length).to eq(0)
```

Comment:
You're [already using a genders file just for this test](https://github.com/alces-software/metalware/pull/147/files#diff-f3cb3abf9ef9f2c5c95bd109c95804a0R32) - can you change this to specify nodes as `01-03` or `01-02`, and then specify the node names directly in the test? You'll need to change some other tests where the same genders file is used to use the same range, but this shouldn't be a complicated change afaics.